pkg/client/record/event.go

```diff
@@ -111,19 +112,19 @@ func (eventBroadcaster *eventBroadcasterImpl) StartRecordingToSink(sink EventSin
 			// Events are safe to copy like this.
 			eventCopy := *event
 			event = &eventCopy
-
+			var patch []byte
 			previousEvent := eventCache.getEvent(event)
 			updateExistingEvent := previousEvent.Count > 0
 			if updateExistingEvent {
-				event.Count = previousEvent.Count + 1
-				event.FirstTimestamp = previousEvent.FirstTimestamp
+				// we still need to copy Name because the Patch relies on the Name to find the target event
 				event.Name = previousEvent.Name
-				event.ResourceVersion = previousEvent.ResourceVersion
+				time, _ := event.LastTimestamp.MarshalJSON()
+				patch = []byte(fmt.Sprintf("{\"count\":%d,\"lastTimestamp\":%s}", previousEvent.Count+1, string(time)))
```

Comment:
There's a function that will produce a patch from two objects, I think.
Look at the PATCH handling code in apiserver, or do a search for
strategicmergepatch.

On Thu, Oct 8, 2015 at 1:07 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/client/record/event.go
> <https://github.com/kubernetes/kubernetes/pull/15268#discussion_r41561782>
> :
>
> >  				event.Name = previousEvent.Name
> > -				event.ResourceVersion = previousEvent.ResourceVersion
> > +				time, _ := event.LastTimestamp.MarshalJSON()
> > +				patch = []byte(fmt.Sprintf("{\"count\":%d,\"lastTimestamp\":%s}", previousEvent.Count+1, string(time)))
>
> I don't know of any common util method for constructing a patch.
> @bgrant0607 <https://github.com/bgrant0607> @lavalamp
> <https://github.com/lavalamp> do you have any suggestion? Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15268/files#r41561782>.
>